crates/primitives/src/receipt.rs

```diff
@@ -44,6 +47,78 @@ impl Receipt {
     }
 }
 
+/// A collection of receipts organized as a two-dimensional vector.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Receipts {
+    /// A two-dimensional vector of optional `Receipt` instances.
+    pub receipt_vec: Vec<Vec<Option<Receipt>>>,
+}
+
+impl Receipts {
+    /// Create a new `Receipts` instance with an empty vector.
+    pub fn new() -> Self {
+        Self { receipt_vec: vec![] }
+    }
+
+    /// Create a new `Receipts` instance from an existing vector.
+    pub fn from_vec(vec: Vec<Vec<Option<Receipt>>>) -> Self {
+        Self { receipt_vec: vec }
+    }
+
+    /// Returns the length of the `Receipts` vector.
+    pub fn len(&self) -> usize {
+        self.receipt_vec.len()
+    }
+
+    /// Returns `true` if the `Receipts` vector is empty.
+    pub fn is_empty(&self) -> bool {
+        self.receipt_vec.is_empty()
+    }
+
+    /// Push a new vector of receipts into the `Receipts` collection.
+    pub fn push(&mut self, receipts: Vec<Option<Receipt>>) {
+        self.receipt_vec.push(receipts);
+    }
+
+    /// Retrieves a vector of references to receipts for the specified index.
+    pub fn root_slow(&self, index: usize) -> Option<Vec<&Receipt>> {
```

Comment:
this is not a root right? wrong fn name?

Reply:
@gakonst It refers to this comment: https://github.com/paradigmxyz/reth/pull/4626#discussion_r1328478497

And this is intended to be used in the following context:
https://github.com/paradigmxyz/reth/blob/24a8590e69e728517e4242e5c98115630426511f/crates/storage/provider/src/bundle_state/bundle_state_with_receipts.rs#L242-L247

Do you prefer something like `receipt_slow`? What do you think @rakita ?

Reply:
You should calculate and return receipt root here, if you return just a list of receipts the name of the function does not make sense.

Reply:
@rakita My bad, I thought I would keep the logic outside then I forgot, I just edited the contents of `root_slow`, it should be fine now